CORE-13356 Client Monitoring: track and capture more connection attributes

pgellert opened the pull request:
This PR introduces tracking a wide set of information about kafka connections that will be surfaced in the `ListKafkaConnections` API. This information will be useful to show information about the behaviour of kafka clients.

You can see here an example output of the API while a producer and a consumer is active: https://gist.githubusercontent.com/pgellert/5705959693f5c8443c54c9f0737d7b8a/raw/62a746362fc8177ac398d590d3af246cb8ffd7e0/gistfile0.txt

See the commit messages for implementation details.

Fixes https://redpandadata.atlassian.net/browse/CORE-13356

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [x] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v25.2.x
- [ ] v25.1.x
- [ ] v24.3.x

## Release Notes
* none
<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->

pgellert committed:
k/server: track uid per-connection

pgellert committed:
k/server: track open_time per-connection

pgellert committed:
k/server: track auth_info per-connection

pgellert committed:
k/server: track last_client_id per-connection

pgellert committed:
k/server: track client_software_* per-connection

pgellert committed:
k/server: track transactional_id per-connection

Track the `transactional_id` on all APIs that have a transactional_id
that identifiers a transactional client.

The `ListTransactions` API is excluded, since it is an admin-style API
that would not correspond to a transactional client.

pgellert committed:
k/server: track group_id per-connection

Track the `group_id` on relevant kafka APIs.

Note that the OffsetDelete API is excluded, as it is considered to be a
kafka admin API and would not identify a consumer group client.

Similarly, the LeaveGroup API is also excluded, as it can be used a
kafka admin API command.

The OffsetFetch API is included if the fetch is for a single group,
because consumer clients typically start with fetching the latest
offsets for their group.

pgellert committed:
k/server: track group_instance_id per-connection

Track the `group_instance_id` on relevant kafka APIs.

The `DescribeGroups` and `LeaveGroup` APIs are excluded as they may be
issued by admin cli tooling.

pgellert committed:
k/server: track group_member_id per-connection

Track the `group_member_id` on relevant kafka APIs.

The `LeaveGroup` API is excluded as it may be used by admin cli tooling.

pgellert committed:
k/server: track api_versions per-connection

This map records, for each Kafka API, the highest version number observed
in requests on this connection. It can be useful for understanding which
protocol versions a client supports or has negotiated with the broker.

pgellert committed:
k/server: track in flight requests per-connection

Tracks the version and in-flight-duration of the currently in-flight
requests, as well as the duration for which the connection has been
idle.

While some of this state is already tracked in
`client_protocol_state::_responses`, it does not currently track the
requests for the full lifetime of the request-response processing, so I
opted to track this state separately. In the future, we could consider
aligning these two states.

pgellert committed:
security: use sasl mechanism_name consistently

Align the `{mechanism}::name` and `sasl_mechanism::mechanism_name()` for
consistency. This was consistent for most mechanisms except for SCRAM
and PLAIN.

`{mechanism}::name` is the kafka-official name, so align on that.

`sasl_mechanism::mechanism_name()` has only had two use cases: (1) used
as a string for pattern matching for audit logging, (2) used for
revoking OIDC credentials. (1) is fixed up in a backward-compatible way,
and (2) is unaffected.

This helps reuse the `{mechanism}::name` strings for pattern matching
into an enum for the `ListKafkaConnections` endpoint.

pgellert committed:
dt: add more fields to list_kafka_connectsion test

Increase the coverage of the smoke test to include more fields.

The `admin_api_require_auth` cluster config setting was removed because
it was unnecessary.

pgellert commented on src/v/kafka/server/connection_context.cc:
```diff
@@ -1101,4 +1184,61 @@ std::ostream& operator<<(std::ostream& o, const virtual_connection_id& id) {
+    if (!new_value) {
+        return;
+    }
+    if (!value || value != new_value) {
```
I don't think this is necessary

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/74066-->
<!--b6b4305f-af07-49ca-a92b-576cb35ef108-->
<!--74916642-be44-4864-b598-c2cd5dd95e37-->
<!--d0759a9e-f381-456d-8370-62a81b6ac550-->
<!--79f67c53-16ed-4caa-b7b4-7b9384e2f65e-->
<!--787c1073-b990-41e1-8f96-acb27123dc66-->
<!--title-->
## Retry command for [Build#74066](https://buildkite.com/redpanda/redpanda/builds/74066)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/datalake/schema_scale_test.py::SchemaScaleTest.schema_scale_test@{"catalog_type":"nessie","cloud_storage_type":1,"query_engine":"trino","use_partition_spec":false}
tests/rptest/tests/datalake/schema_scale_test.py::SchemaScaleTest.schema_scale_test@{"catalog_type":"rest_hadoop","cloud_storage_type":1,"query_engine":"trino","use_partition_spec":false}
tests/rptest/tests/datalake/schema_scale_test.py::SchemaScaleTest.schema_scale_test@{"catalog_type":"rest_hadoop","cloud_storage_type":1,"query_engine":"spark","use_partition_spec":false}
tests/rptest/tests/datalake/schema_scale_test.py::SchemaScaleTest.schema_scale_test@{"catalog_type":"nessie","cloud_storage_type":1,"query_engine":"spark","use_partition_spec":false}
tests/rptest/tests/list_kafka_connections_test.py::AdminV2ListKafkaConnectionsTest.test_list_kafka_connections
tests/rptest/tests/datalake/schema_scale_test.py::SchemaScaleTest.schema_scale_test@{"catalog_type":"rest_jdbc","cloud_storage_type":1,"query_engine":"spark","use_partition_spec":false}
```
<!--/body-->

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--1a8ccf91-e333-49a1-9c20-2ec86fd35240-->
<!--TEST RESULTS-->
<!--e14bf8ec-ae20-4c00-a9a0-0c1b83b4a8b2-->
<!--TEST RESULTS-->
<!--96d36cc4-cd4b-4f59-890c-2d4c881494dc-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/74066">build#74066</a></summary>


| test_class                                         | test_method                                        | test_arguments                                                                                                 | test_kind   | job_url                                                                                   | test_status   | passed   | reason   | test_history                                                                                                                                                              |
|:---------------------------------------------------|:---------------------------------------------------|:---------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:---------|:--------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| DatalakeManyTopicsTest                             | test_basic                                         | {"cloud_storage_type": 1}                                                                                      | integration | https://buildkite.com/redpanda/redpanda/builds/74066#0199dd7d-9aa6-42a2-a078-5d202b9e476c | FAIL          | 0/1      |          | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DatalakeManyTopicsTest&test_method=test_basic                           |
| SchemaScaleTest                                    | schema_scale_test                                  | {"catalog_type": "nessie", "cloud_storage_type": 1, "query_engine": "spark", "use_partition_spec": false}      | integration | https://buildkite.com/redpanda/redpanda/builds/74066#0199dd7d-9aa0-41e8-9cb5-9ed068bed6b6 | FAIL          | 0/1      |          | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=SchemaScaleTest&test_method=schema_scale_test                           |
| SchemaScaleTest                                    | schema_scale_test                                  | {"catalog_type": "rest_hadoop", "cloud_storage_type": 1, "query_engine": "spark", "use_partition_spec": false} | integration | https://buildkite.com/redpanda/redpanda/builds/74066#0199dd7d-9aa1-40f2-8414-47ea6228e4a5 | FAIL          | 0/1      |          | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=SchemaScaleTest&test_method=schema_scale_test                           |
| SchemaScaleTest                                    | schema_scale_test                                  | {"catalog_type": "rest_jdbc", "cloud_storage_type": 1, "query_engine": "spark", "use_partition_spec": false}   | integration | https://buildkite.com/redpanda/redpanda/builds/74066#0199dd7d-9aa2-4f3c-b193-7d72b3c89759 | FAIL          | 0/1      |          | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=SchemaScaleTest&test_method=schema_scale_test                           |
| SchemaScaleTest                                    | schema_scale_test                                  | {"catalog_type": "nessie", "cloud_storage_type": 1, "query_engine": "trino", "use_partition_spec": false}      | integration | https://buildkite.com/redpanda/redpanda/builds/74066#0199dd7d-9aa3-4f8c-b78e-a24861b745e1 | FAIL          | 0/1      |          | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=SchemaScaleTest&test_method=schema_scale_test                           |
| SchemaScaleTest                                    | schema_scale_test                                  | {"catalog_type": "rest_hadoop", "cloud_storage_type": 1, "query_engine": "trino", "use_partition_spec": false} | integration | https://buildkite.com/redpanda/redpanda/builds/74066#0199dd7d-9aa5-4f9b-94fc-4af1e8ef5d12 | FAIL          | 0/1      |          | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=SchemaScaleTest&test_method=schema_scale_test                           |
| SchemaScaleTest                                    | schema_scale_test                                  | {"catalog_type": "rest_jdbc", "cloud_storage_type": 1, "query_engine": "trino", "use_partition_spec": false}   | integration | https://buildkite.com/redpanda/redpanda/builds/74066#0199dd7d-9aa6-42a2-a078-5d202b9e476c | FAIL          | 0/1      |          | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=SchemaScaleTest&test_method=schema_scale_test                           |
| AdminV2ListKafkaConnectionsTest                    | test_list_kafka_connections                        | null                                                                                                           | integration | https://buildkite.com/redpanda/redpanda/builds/74066#0199dd7d-9aa2-4f3c-b193-7d72b3c89759 | FAIL          | 0/1      |          | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=AdminV2ListKafkaConnectionsTest&test_method=test_list_kafka_connections |
| src/v/cloud_storage/tests/segment_meta_cstore_test | src/v/cloud_storage/tests/segment_meta_cstore_test |                                                                                                                | unit        | https://buildkite.com/redpanda/redpanda/builds/74066#0199dd4e-98e5-44fc-a07a-fb645df052ea | FAIL          | 0/1      |          |                                                                                                                                                                           |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/74153">build#74153</a></summary>


| test_class                          | test_method               | test_arguments                                                                                                 | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                         | test_history                                                                                                                                                     |
|:------------------------------------|:--------------------------|:---------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------|
| DatalakeManyTopicsTest              | test_basic                | {"cloud_storage_type": 1}                                                                                      | integration | https://buildkite.com/redpanda/redpanda/builds/74153#0199e2b4-40d8-492e-aa08-e6b2759ffb00 | FAIL          | 0/1      |                                                                                                                                                                | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DatalakeManyTopicsTest&test_method=test_basic                  |
| DatalakeManyTopicsTest              | test_basic                | {"cloud_storage_type": 1}                                                                                      | integration | https://buildkite.com/redpanda/redpanda/builds/74153#0199e2ba-f5d6-493e-a78a-a27a48d5def6 | FAIL          | 0/1      |                                                                                                                                                                | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DatalakeManyTopicsTest&test_method=test_basic                  |
| SchemaScaleTest                     | schema_scale_test         | {"catalog_type": "nessie", "cloud_storage_type": 1, "query_engine": "spark", "use_partition_spec": false}      | integration | https://buildkite.com/redpanda/redpanda/builds/74153#0199e2b4-40d4-4ba8-8493-8e4d4a037f4a | FAIL          | 0/1      |                                                                                                                                                                | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=SchemaScaleTest&test_method=schema_scale_test                  |
| SchemaScaleTest                     | schema_scale_test         | {"catalog_type": "rest_hadoop", "cloud_storage_type": 1, "query_engine": "spark", "use_partition_spec": false} | integration | https://buildkite.com/redpanda/redpanda/builds/74153#0199e2b4-40d5-44db-901f-f8251870f31b | FAIL          | 0/1      |                                                                                                                                                                | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=SchemaScaleTest&test_method=schema_scale_test                  |
| SchemaScaleTest                     | schema_scale_test         | {"catalog_type": "rest_jdbc", "cloud_storage_type": 1, "query_engine": "spark", "use_partition_spec": false}   | integration | https://buildkite.com/redpanda/redpanda/builds/74153#0199e2b4-40d5-45f9-ba75-5027ad6aee30 | FAIL          | 0/1      |                                                                                                                                                                | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=SchemaScaleTest&test_method=schema_scale_test                  |
| SchemaScaleTest                     | schema_scale_test         | {"catalog_type": "rest_jdbc", "cloud_storage_type": 1, "query_engine": "spark", "use_partition_spec": false}   | integration | https://buildkite.com/redpanda/redpanda/builds/74153#0199e2ba-f5d0-46df-aaa7-0233c3ac2140 | FAIL          | 0/1      |                                                                                                                                                                | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=SchemaScaleTest&test_method=schema_scale_test                  |
| SchemaScaleTest                     | schema_scale_test         | {"catalog_type": "nessie", "cloud_storage_type": 1, "query_engine": "trino", "use_partition_spec": false}      | integration | https://buildkite.com/redpanda/redpanda/builds/74153#0199e2b4-40d7-469d-b5c9-7a934c3488eb | FAIL          | 0/1      |                                                                                                                                                                | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=SchemaScaleTest&test_method=schema_scale_test                  |
| SchemaScaleTest                     | schema_scale_test         | {"catalog_type": "rest_hadoop", "cloud_storage_type": 1, "query_engine": "trino", "use_partition_spec": false} | integration | https://buildkite.com/redpanda/redpanda/builds/74153#0199e2b4-40d8-4245-badd-4eb14c9a1f7f | FAIL          | 0/1      |                                                                                                                                                                | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=SchemaScaleTest&test_method=schema_scale_test                  |
| SchemaScaleTest                     | schema_scale_test         | {"catalog_type": "rest_jdbc", "cloud_storage_type": 1, "query_engine": "trino", "use_partition_spec": false}   | integration | https://buildkite.com/redpanda/redpanda/builds/74153#0199e2b4-40d8-492e-aa08-e6b2759ffb00 | FAIL          | 0/1      |                                                                                                                                                                | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=SchemaScaleTest&test_method=schema_scale_test                  |
| FlinkBasicTests                     | test_transaction_workload | null                                                                                                           | integration | https://buildkite.com/redpanda/redpanda/builds/74153#0199e2ba-f5c9-40fc-b88c-50e6d714b83f | FLAKY         | 3/21     | upstream reliability is '100.0'. current run reliability is '14.285714285714285'. drift is 85.71429 and the allowed drift is set to 50. The test should FAIL   | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=FlinkBasicTests&test_method=test_transaction_workload          |
| RpkDebugBundleTest                  | test_debug_bundle         | null                                                                                                           | integration | https://buildkite.com/redpanda/redpanda/builds/74153#0199e2b4-40d8-492e-aa08-e6b2759ffb00 | FAIL          | 0/1      |                                                                                                                                                                | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=RpkDebugBundleTest&test_method=test_debug_bundle               |
| WriteCachingFailureInjectionE2ETest | test_crash_all            | {"use_transactions": false}                                                                                    | integration | https://buildkite.com/redpanda/redpanda/builds/74153#0199e2ba-f5d9-4344-872c-efaece0715af | FLAKY         | 13/21    | upstream reliability is '82.8125'. current run reliability is '61.904761904761905'. drift is 20.90774 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=WriteCachingFailureInjectionE2ETest&test_method=test_crash_all |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/74249">build#74249</a></summary>


| test_class                    | test_method                    | test_arguments   | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 | test_history                                                                                                                                                               |
|:------------------------------|:-------------------------------|:-----------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|:---------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ShadowLinkingReplicationTests | test_replication_with_failures | null             | integration | https://buildkite.com/redpanda/redpanda/builds/74249#0199e83e-fade-4db1-89c9-5024fcebfbec | FLAKY         | 20/21    | upstream reliability is '100.0'. current run reliability is '95.23809523809523'. drift is 4.7619 and the allowed drift is set to 50. The test should PASS              | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ShadowLinkingReplicationTests&test_method=test_replication_with_failures |
| TimeQueryTest                 | test_timequery_with_local_gc   | null             | integration | https://buildkite.com/redpanda/redpanda/builds/74249#0199e83e-fae0-4f24-a3ff-3cac6008cfeb | FLAKY         | 17/21    | upstream reliability is '90.15817223198594'. current run reliability is '80.95238095238095'. drift is 9.20579 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=TimeQueryTest&test_method=test_timequery_with_local_gc                   |
</details>
<!--/body-->

pgellert committed:
k/server: make get_principal() safe pre-authn

Ensure that `connection_context::get_principal()` is safe to call before
sasl authentication completes. During that time, `get_principal()` now
returns the anonymous user.

The current code paths all call `get_principal()` only after
authentication completes, however, to allow reusing the
`get_principal()` helper for the ListKafkaConnections admin API, we
extend it here to make the logic safe pre-authentication completion as
well. This is preferred over reimplementing `get_principal()` in the
`to_proto()` handler for better code reuse.

pgellert commented:
[force-push](https://github.com/redpanda-data/redpanda/compare/14122a5b0cc581c23cd48a6a73540414b4fbfe95..45ba5c0ce2a473d7410a21469a2ea910290e0773): fix test-highlighted UB when `get_principal()` accessed mid-authentication

BenPope commented on src/v/kafka/server/connection_context.cc:
```diff
@@ -823,6 +823,7 @@ proto::admin::kafka_connection connection_context::to_proto() const {
     res.set_shard_id(ss::this_shard_id());
     res.set_node_id(
       config::node().node_id.value().value_or(model::unassigned_node_id));
+    res.set_uid(fmt::format("{}", _attributes.connection_id));
```
nit: `ssx::sformat` should avoid a copy.

BenPope commented on src/v/kafka/server/connection_context.cc:
```diff
@@ -1160,4 +1167,13 @@ std::ostream& operator<<(std::ostream& o, const virtual_connection_id& id) {
+    }
+    if (!value || value != new_value) {
+        value = ss::sstring{*new_value};
+    }
```
nit:
```suggestion
    if (new_value && value != *new_value) {
        value.emplace(*new_value);
    }
```

BenPope commented on src/v/kafka/server/handlers/api_versions.cc:
```diff
@@ -123,6 +123,11 @@ api_versions_response api_versions_handler::handle_raw(request_context& ctx) {
+        ctx.connection()->attributes().last_client_software_name.update(
+          request.data.client_software_name);
+        ctx.connection()->attributes().last_client_software_version.update(
+          request.data.client_software_version);
```
These are empty for `api_version` <v3, did you want to leave the `optional` `nullopt`, or empty string?

pgellert replied:
It doesn't change the `ListKafkaConnections` output because in either case the fields are surfaced as an empty string. However, yeah it's probably nicer to wrap this in a version check to at least document that they are only present in later versions. Adding that now.

BenPope commented on src/v/kafka/server/connection_context.h:
```diff
@@ -181,6 +181,31 @@ struct connection_attributes {
+            api_key key;
+            clock::time_point recv_time;
+        };
+        using queue_t = std::deque<in_flight_request>;
```
We do have a `chunked_circular_buffer`, which is just an alias to `std::deque`.

pgellert replied:
So it sounds like I should just use `std::deque`. Anything I am missing here?

BenPope replied:
> So it sounds like I should just use `std::deque`. Anything I am missing here?

Nothing to do here.

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/74136-->
<!--abddf328-b41a-41e8-99df-7748455fae3c-->
<!--title-->
## Retry command for [Build#74136](https://buildkite.com/redpanda/redpanda/builds/74136)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/datalake/schema_scale_test.py::SchemaScaleTest.schema_scale_test@{"catalog_type":"nessie","cloud_storage_type":1,"query_engine":"trino","use_partition_spec":false}
```
<!--/body-->

pgellert committed:
k/server: prefer ssx::sformat in to_proto

To avoid a std::string to ss::sstring conversion

pgellert commented:
[force-push](https://github.com/redpanda-data/redpanda/compare/45ba5c0ce2a473d7410a21469a2ea910290e0773..86f6a879974657e3b32a498504db5f3b05806cf4): address code review feedback

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/74144-->
<!--18a84b8c-46e9-4f1e-972d-eb69f97637de-->
<!--title-->
## Retry command for [Build#74144](https://buildkite.com/redpanda/redpanda/builds/74144)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/datalake/schema_scale_test.py::SchemaScaleTest.schema_scale_test@{"catalog_type":"nessie","cloud_storage_type":1,"query_engine":"trino","use_partition_spec":false}
```
<!--/body-->

pgellert commented:
/ci-repeat 1

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/74153-->
<!--00479812-129a-46e0-bc2f-ad282263af50-->
<!--d5582d93-7743-49f7-985c-7b2f0bdfb689-->
<!--d171a13d-ce10-4cf7-8eb1-2ebd73b301a9-->
<!--9475cbf8-f31a-4913-94b9-b1a0b0154a67-->
<!--94a111e0-6dbe-45ff-bc00-531fd79953ee-->
<!--51ac40e1-1a0e-4cfc-9adf-16df07121e30-->
<!--8d400dc7-43ae-42b9-8eb1-1f5452eaad5e-->
<!--title-->
## Retry command for [Build#74153](https://buildkite.com/redpanda/redpanda/builds/74153)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/datalake/schema_scale_test.py::SchemaScaleTest.schema_scale_test@{"catalog_type":"nessie","cloud_storage_type":1,"query_engine":"trino","use_partition_spec":false}
tests/rptest/tests/datalake/schema_scale_test.py::SchemaScaleTest.schema_scale_test@{"catalog_type":"nessie","cloud_storage_type":1,"query_engine":"spark","use_partition_spec":false}
tests/rptest/tests/datalake/schema_scale_test.py::SchemaScaleTest.schema_scale_test@{"catalog_type":"rest_hadoop","cloud_storage_type":1,"query_engine":"trino","use_partition_spec":false}
tests/rptest/tests/datalake/schema_scale_test.py::SchemaScaleTest.schema_scale_test@{"catalog_type":"rest_jdbc","cloud_storage_type":1,"query_engine":"spark","use_partition_spec":false}
tests/rptest/tests/datalake/schema_scale_test.py::SchemaScaleTest.schema_scale_test@{"catalog_type":"rest_hadoop","cloud_storage_type":1,"query_engine":"spark","use_partition_spec":false}
tests/rptest/tests/flink_basic_test.py::FlinkBasicTests.test_transaction_workload
```
<!--/body-->

pgellert commented:
[force-push](https://github.com/redpanda-data/redpanda/compare/86f6a879974657e3b32a498504db5f3b05806cf4..9592ce8d8fb0780ec44cdb6bc248e21ac628fb6d): fix the logic of the in flight request tracker and add a unit test for it
